packages/core/saved-objects/core-saved-objects-base-server-internal/src/saved_objects_config.ts

```diff
@@ -50,19 +52,20 @@ export const savedObjectsConfig: ServiceConfigDescriptor<SavedObjectsConfigType>
   path: 'savedObjects',
   schema: soSchema,
 };
-
 export class SavedObjectConfig {
   public maxImportPayloadBytes: number;
   public maxImportExportSize: number;
-
+  public allowHttpApiAccess: boolean; // depend on env: see https://github.com/elastic/dev/issues/2200
   public migration: SavedObjectsMigrationConfigType;
 
   constructor(
     rawConfig: SavedObjectsConfigType,
-    rawMigrationConfig: SavedObjectsMigrationConfigType
+    rawMigrationConfig: SavedObjectsMigrationConfigType,
+    allowHttpApiAccess: boolean
```

Comment:
Naming is hard and `hideFromHttpApis` or `denyHttpApiAccess` sound rather negative, hence the more "positive" choice.